internal/testing/e2e/consul.go

```diff
@@ -530,6 +540,33 @@ func CreateConsulAuthMethod(namespace string) env.Func {
 	}
 }
 
+func IsEnterprise() bool {
+	return strings.HasSuffix(consulImage, "ent")
```

Comment:
This feels like a kinda sketchy check, but fine for now. I'm curious if there's a better to do something like query supported features.

Reply:
Yeah, it's definitely a hack, and not proud of it. Main alternative I can think of is doing something like just adding a `consul_ent.go` with a build tag that overrides `IsEnterprise` or something of the sort and add a `go test -tag ent` to pick it up. Alternatively I'd love if Consul had the equivalent of a `version` api or something of the sort that gave you back whether or not you had an enterprise deployment or not, but I don't think there is, only some other gross hacks like testing for the existence of APIs

Reply:
Yeah, not that I don't have a conflict of interest, but am kind of partial to something like what Elasticsearch does, exposing version info at a top-level endpoint. Assuming you have auth turned on this could be protected to avoid folks banner grabbing:

```bash
➜  ~ curl http://localhost:9200
{
  "name" : "f9b9188b4715",
  "cluster_name" : "docker-cluster",
  "cluster_uuid" : "PU5ca1ZFRBGuXmYmCPlE9A",
  "version" : {
    "number" : "8.0.0",
    "build_flavor" : "default",
    "build_type" : "docker",
    "build_hash" : "1b6a7ece17463df5ff54a3e1302d825889aa1161",
    "build_date" : "2022-02-03T16:47:57.507843096Z",
    "build_snapshot" : false,
    "lucene_version" : "9.0.0",
    "minimum_wire_compatibility_version" : "7.17.0",
    "minimum_index_compatibility_version" : "7.0.0"
  },
  "tagline" : "You Know, for Search"
}
```